lib/src/Base/Type/IndicesFrozenCollection.cxx

```diff
@@ -0,0 +1,137 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief IndicesFrozenCollection stores a Collection of Indices contiguously.
+ *
+ *  Copyright 2005-2018 Airbus-EDF-IMACS-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/IndicesFrozenCollection.hxx"
+#include "openturns/IndicesFrozenCollectionImplementation.hxx"
+#include "openturns/IndicesFrozenCollectionStrided.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(IndicesFrozenCollection)
+
+/* Default constructor */
+IndicesFrozenCollection::IndicesFrozenCollection()
+  : TypedInterfaceObject<IndicesFrozenCollectionImplementation>(new IndicesFrozenCollectionImplementation())
+{
+  // Nothing to do
+}
+
+namespace {
+Bool isStrided(const Collection<Indices> & values)
+{
+  if (values.getSize() <= 1) return true;
+  const UnsignedInteger firstStride = values[0].getSize();
+  for(Collection<Indices>::const_iterator it = values.begin(), guard = values.end(); it != guard; ++it)
+  {
+    if (firstStride != it->getSize())
+      return false;
+  }
+  return true;
+}
+}
```

Comment:
Just a warning here, we use unity builds (we concatenate several .cxx in order to reduce build time) on the CIs, and functions in anonymous namespace might be merged from another .cxx file, so we have to keep in mind not to create conflicts.

Reply:
Indeed, I'll prefix this function, thanks.